umapi_client/functional.py

```diff
@@ -329,39 +317,32 @@ def maybe_split_groups(self, max_groups):
         new_commands = []
         # return True if we split at least once
         maybe_split = False
+        valid_step_keys = ['add', 'addRoles', 'remove']
         for command in self.commands:
+            # commands are assumed to contain a single key
             step_key, step_args = next(six.iteritems(command))
-            if step_key not in [StepKeys.add.name, StepKeys.remove.name, StepKeys.addRoles.name]:
+            if step_key not in valid_step_keys:
+                new_commands.append(command)
                 continue
-            split_commands = self._split_groups(step_key, command, max_groups)
-            if len(split_commands) > 1:
-                maybe_split = True
+            split_commands = [command]
+            while True:
+                new_command = {step_key: {}}
+                if not isinstance(command[step_key], dict):
+                    break
+                for group_type, groups in six.iteritems(command[step_key]):
+                    if len(groups) > max_groups:
+                        maybe_split = True
```

Comment:
Move this down to line 338 (see comment there).  It's always better not to do redundant stuff in inner loops, and keep them as clean as possible.